### PR TITLE
Add pagination to the resources page

### PR DESCRIPTION
**Pull request by anthonydillon**
## Done
Add pagination to the resources page

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: [http://0.0.0.0:8001/resources](http://0.0.0.0:8001/resources)
- Check out the pagination at the bottom of the items
- Play around with the filtering and check it all works as expected

## Issue / Card
Fixes https://github.com/ubuntudesign/web-squad/issues/289

### CODE REVIEW

File: webapp/views.py [hunk 1]
```diff
@@ -72,72 +75,142 @@ def _normalise_resources(self, resources):
...
-                    limit=3,
+                posts = get_json_feed_content(api_url, limit=3)
+                feed_items['pagination'] = self._generate_pagination_queries(
+                    page,
+                    len(posts)
                 )
+                feed_items['posts'] = {}
+                feed_items['posts'][group_name] = {}
+                feed_items['posts'][group_name]['posts'] = posts
                 feed_items[group_name]['group_name'] = group['name']
```
Comment (Caleb-Ellis) [inline webapp/views.py]: I'm getting a KeyError in python from this line of code

File: webapp/views.py [hunk 2]
```diff
@@ -72,72 +75,142 @@ def _normalise_resources(self, resources):
...
+                feed_items['posts'][topic]['posts'] = posts
+                feed_items['posts'][topic]['group_name'] = title
         else:
             for group_name, group in self.GROUPS.items():
                 api_url = (
                     '{api_url}/posts'
-                    '?group={group_id}&{resource_filter}'
+                    '?group={group_id}&{resource_filter}&per_page={per_page}'
+                    '&page={page}'
                 ).format(
```
Comment (tbille) [inline webapp/views.py]: Not from your PR, but we need to choose.
```Python
api_url = (...).format(...)

or

api_url = (...)
api_url = api_url.format(...)
```

**Comment by frankiegranato** [comment]
@anthonydillon I can't review this because the demo is broken :-(

File: webapp/views.py (hunk 1 above)
Comment (anthonydillon) [reply webapp/views.py]: Fixed

**Comment by anthonydillon** [comment]
@Caleb-Ellis Fixed up the issues, ready for another review

@tbille I have gone for `api_url = (...).format(...)`

File: webapp/views.py (hunk 2 above)
Comment (WillMoggridge) [reply webapp/views.py]: I generally like it in the first style and like this PR.

Most of the time, the latter option just seems to add more for no gain.

**Comment by Caleb-Ellis** [comment]
## QA
- Checked that pagination worked as expected on /resources for All, Cloud and server, Desktop, Internet of Things
- Confirmed that on page 1 there is no previous page button
- Confirmed that once the article limit is reached, there is no next page button

**Pull request merged**